### PR TITLE
[KOGITO-851] Adding event support to Jobs Service

### PR DESCRIPTION
**Pull request by ricardozanini**
Signed-off-by: Ricardo Zanini <zanini@redhat.com>

See:
https://issues.redhat.com/browse/KOGITO-851

Depends on:
https://github.com/kiegroup/kogito-images/pull/114

Image for testing:
`quay.io/ricardozanini/kogito-jobs-service:0.9.0-rc1`

In this PR we introduce the Kafka integration with Jobs Service. When any parameter is set (URL, service instance or `useKogitoInfra`), eventing is automatically enabled in the Jobs Service image.

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: pkg/controller/kogitojobsservice/kogitojobsservice_controller.go [hunk 1]
```diff
@@ -143,8 +144,13 @@ const (
 	maxIntervalLimitRetryEnvKey       = "MAX_INTERVAL_LIMIT_RETRY"
 	backOffRetryDefaultValue          = 1000
 	maxIntervalLimitRetryDefaultValue = 60000
+	kafkaTopicNameJobsEvents          = "kogito-job-service-job-status-events"
```
Comment (radtriste) [inline pkg/controller/kogitojobsservice/kogitojobsservice_controller.go]: Are you sure about this name ?
Shouldn't we rely on that topic name ?
https://github.com/kiegroup/kogito-apps/blob/7a53dc2f6e2e15e6dcc2ed194e043fc29aee1226/data-index/data-index-service/src/main/resources/application.properties#L42
Also present here: https://github.com/kiegroup/kogito-cloud-operator/blob/c3a80921e3271ea5bf0c28ee3cfb297610bc84c5/pkg/controller/kogitodataindex/kogitodataindex_controller.go#L166
Or do I miss something ?

Comment (ricardozanini) [reply pkg/controller/kogitojobsservice/kogitojobsservice_controller.go]: This is the name presented in the properties, this name is used to set the env var:
https://github.com/kiegroup/kogito-apps/blob/master/jobs-service/src/main/resources/application.properties#L74

Comment (ricardozanini) [reply pkg/controller/kogitojobsservice/kogitojobsservice_controller.go]: In the Data Index the name differ:
https://github.com/kiegroup/kogito-apps/blob/master/data-index/data-index-service/src/main/resources/application.properties#L42

Comment (radtriste) [reply pkg/controller/kogitojobsservice/kogitojobsservice_controller.go]: ok. my bad. indeed did not look into the correct app ...

**Pull request merged**